File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Tasks/managingduration.rb
module Asposetasksjava
  module ManagingDuration
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new

        task = project.getRootTask().getChildren().add("Task")

        tsk = Rjb::import('com.aspose.tasks.Tsk')

		# task duration in days (default time unit)
		duration = task.get(tsk.DURATION)
		puts "Duration in Days: " + duration.toString()

		# convert to hours time unit
		duration = duration.convert(Rjb::import('com.aspose.tasks.TimeUnitType').Hour)
		puts "Duration in Hours: "+ duration.toString()

		# Decrease task duration
		task.set(tsk.DURATION, task.get(tsk.DURATION).subtract(0.5))
		puts "0.5 wks: " + task.get(tsk.DURATION).toString()
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithTasks/ManagingDuration.php
<?php
namespace Aspose\Tasks\WorkingWithTasks;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\Tsk as Tsk;
use com\aspose\tasks\TimeUnitType as TimeUnitType;
class ManagingDuration{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project();

        $task = $project->getRootTask()->getChildren()->add("Task");

        $tsk = new Tsk();

		# task duration in days (default time unit)
		$duration = $task->get($tsk->DURATION);
		print "Duration in Days: " . $duration->toString();

		# convert to hours time unit
        $timeUnitType=new TimeUnitType();
		$duration = $duration->convert($timeUnitType->Hour);
		print "Duration in Hours: ". (string)$duration.PHP_EOL;//.toString();

		# Decrease task duration
		$task->set($tsk->DURATION, $task->get($tsk->DURATION)->subtract(0.5));
		print "0.5 wks: " . (string)$task->get($tsk->DURATION).PHP_EOL;//.toString();
    }

}
?>